Devex-1017: build-client-integration test intermittent failures

rosiehsmith opened the pull request:
[Associated trello card
](https://trello.com/c/dgva54sP/1017-investigate-circle-build-inconsistency-integration-tests-random-failure)

This PR (1/x) specifically addresses the integration test `web-client/integration-tests/noticeOfChangeOfAddressQCJourney.test.js` like in this failed[ CircleCI workflow ](https://app.circleci.com/pipelines/github/ustaxcourt/ef-cms/3273/workflows/a9e97ff5-7321-47a4-9f16-2f13d9f4a115/jobs/33095) as well as [noticeOfTrialSessionWithPaperService.test.js](https://app.circleci.com/pipelines/github/ustaxcourt/ef-cms/3279/workflows/f915f318-811e-4176-9a50-91d0a0a31ac8/jobs/33207) and [noticeOfTrialSessionWithElectronicService.test.js](https://app.circleci.com/pipelines/github/ustaxcourt/ef-cms/3290/workflows/2d82b0b0-7cd5-4d90-9654-3cfa1f7e1b56/jobs/33363).

**Suspected cause of intermittency**: now that we are using `isAsync` on our async endpoints, the tests are not waiting until they're done processing and immediately return a 204

**Changes**: We're now waiting for specific items other than the loading spinner, like the progress bar to complete and disappear when updating the practitioner address which addresses the async `updateUserContactInformationLambda`. For noticeOfTrialSessions tests, we're waiting for the correct page and alertSuccess.

Also noticed an import was wrong in `web-client/src/presenter/sequences/loadMorePendingItemsSequence.js` so fixed that since it failed while building the docker image created to test these locally.

rachelschneiderman committed:
Devex 1017: Increase timeout in helper method that waits for loading icon to hide to be equal to jest test timeout

rachelschneiderman committed:
Devex 1017: Only run integration tests on exp

rosiehsmith committed:
devex-1017: create a docker image for running intermittently failing build-client-integration tests; fix path for a sequence [skip ci]

rachelschneiderman committed:
Devex 1017: Upgrade ES and try waiting for indexing before returning

rachelschneiderman committed:
Merge branch 'experimental2' of https://github.com/flexion/ef-cms into experimental2

rachelschneiderman committed:
Devex 1017: Upgrade ES docker container

rachelschneiderman committed:
Revert "Devex 1017: Upgrade ES and try waiting for indexing before returning"

This reverts commit 05b4244afc01ee8ce1d08b9e84e626fb1672dcee.

rachelschneiderman committed:
Revert "Devex 1017: Upgrade ES docker container"

This reverts commit 3714a5b3ae113969fdb04cabe2e4e718df0cd124.

rosiehsmith committed:
devex-1017: adding a waitForProgressBarToHide helper method on changing address [skip ci]

rachelschneiderman committed:
Devex 1017: Increase wait for progress bar to be hidden, add loggin [skip ci]

rosiehsmith committed:
devex-1017: splitting out test runs to be more similar to circleCI container runs with frequently intermittent tests [skip ci]

rachelschneiderman committed:
Devex 1017: Reduce wait time

rachelschneiderman committed:
Merge branch 'experimental2' of https://github.com/flexion/ef-cms into experimental2

rosiehsmith committed:
devex-1017: cleanup

rosiehsmith committed:
devex-1017: make some stuff more clear with .md text

mrinsin commented on docs/testing.md:
````diff
@@ -275,4 +275,12 @@ All of the expected output images are found in the `./shared/test-pdf-expected-i
 cp -r shared/test-output/*.1.png shared/test-pdf-expected-images/
+```
+
+## Client Integration Testing
````
This is awesome!

mrinsin reviewed (approved):
![alt text](https://c.tenor.com/cLpcad-Dyh4AAAAC/wow-amazed.gif)

rosiehsmith committed:
devex-1017: make waitFor method more generic to pass in item

# Conflicts:
#	web-client/integration-tests/helpers.js

rosiehsmith committed:
devex-1017: reuse waitForLoadingComponentToHide with default args

# Conflicts:
#	web-client/integration-tests/helpers.js
#	web-client/integration-tests/journey/petitionsClerkCompletesAndSetsTrialSession.js

rosiehsmith committed:
devex-1017: update missed waitForLoadingComponentToHide

rosiehsmith committed:
devex-1017: update helper to expect item not page - fix intermittency with petitionsClerkCompletesAndSetsTrialSession for Generate Notices of Trial Session tests both paper and electronic service

# Conflicts:
#	web-client/integration-tests/helpers.js
#	web-client/integration-tests/journey/petitionsClerkCompletesAndSetsTrialSession.js

rosiehsmith committed:
devex-1017: wait for PrintPaperTrialNotices page after setTrialSessionCalendarSequence

TomElliottFlexion commented:
https://media.giphy.com/media/QZgzJIU5ODel9D1Hob/giphy.gif

TomElliottFlexion commented:
![alt text](https://media.giphy.com/media/QZgzJIU5ODel9D1Hob/giphy.gif)

halprin reviewed (approved):
I'm late but this is great!
